mbuild/formats/lammpsdata.py

```diff
@@ -610,16 +731,27 @@ def write_lammpsdata(
                     )
                 }
                 for params, idx in sorted_bond_types.items():
-                    data.write(
-                        "{}\t{}\t\t{}\t\t# {}\t{}\n".format(
-                            idx,
-                            params[0],
-                            params[1],
-                            params[2][0],
-                            params[2][1],
+                    # If the user specified LJ unit style, revert the internal conversion of k by ParmEd
+                    if unit_style == "lj":
+                        data.write(
```

Comment:
It seems like the  scaling factors are applied when calling `_get_bond_types` function? Why is the unit conversion necessary here?

Reply:
Scaling factors are applied after sorting bond types and just before the bond coefficients and distances are written to the data file *if and only if* the user has selected the "lj" reduced unit style. 

This was an easier fix because we did not want to update the parameters (i.e. k and req) within unique_bond_types unless the user has selected reduced units. So, we opted to do the conversion just before writing rather than updating the original array containing the parameters.

Reply:
but it seems like the conversion already take place [here](https://github.com/mosdef-hub/mbuild/blob/5608ec0afc0779e9ce9455b9bbd1d81931438ac5/mbuild/formats/lammpsdata.py#L1032-L1043)? 
```
def _get_bond_types(
    structure, bonds, sigma_conversion_factor, epsilon_conversion_factor
):
    unique_bond_types = dict(
        enumerate(
            set(
                [
                    (
                        round(
                            bond.type.k
                            * (
                                sigma_conversion_factor ** 2
                                / epsilon_conversion_factor
                            ),
                            3,
                        ),
                        round(bond.type.req / sigma_conversion_factor, 3),
                        tuple(sorted((bond.atom1.type, bond.atom2.type))),
                    )
                    for bond in structure.bonds
                ]
            )
        )
    )
    unique_bond_types = OrderedDict(
        [(y, x + 1) for x, y in unique_bond_types.items()]
    )
    bond_types = [
        unique_bond_types[
            (
                round(
                    bond.type.k
                    * (
                        sigma_conversion_factor ** 2 / epsilon_conversion_factor
                    ),
                    3,
                ),
                round(bond.type.req / sigma_conversion_factor, 3),
                tuple(sorted((bond.atom1.type, bond.atom2.type))),
            )
        ]
        for bond in structure.bonds
    ]
    return bond_types, unique_bond_types
```

Reply:
same for angles and dihedrals